plutus-core/plutus-ir/src/PlutusIR/Transform/Inline.hs

```diff
@@ -309,16 +335,17 @@ processSingleBinding body = \case
     -- Just process all the subterms
     b -> Just <$> forMOf bindingSubterms b processTerm
 
--- NOTE:  Nothing means that we are inlining the term:
+-- Check against the heuristics we have for inlining and either inline the term binding or not.
+-- Nothing means that we are inlining the term:
 --   * we have extended the substitution, and
---   * we are removing the binding (hence we return Nothing)
+--   * we are removing the binding (hence we return Nothing).
 maybeAddSubst
     :: forall tyname name uni fun a. InliningConstraints tyname name uni fun
-    => Term tyname name uni fun a
-    -> a
-    -> Strictness
-    -> name
-    -> Term tyname name uni fun a
+    => Term tyname name uni fun a -- ^ The body of the let binding that is to be inlined.
```

Comment:
```suggestion
    => Term tyname name uni fun a -- ^ The body of the let term that is to be inlined.
```

Reply:
There's a confusion here: we are processing one _binding_ from a given _let term_. The body is the body of the _let term_, the other bits are the pieces of the _binding_. The comments make it sound like they're all associated with the same structure, which they're not.